partner_firstname/models.py

```diff
@@ -38,12 +38,16 @@ class ResPartner(models.Model):
         required=False,
         store=True)
 
+    @api.model
+    def _get_computed_name(self, lastname, firstname):
+        """Write the 'name' field according to splitted data."""
+        return u" ".join((p for p in (lastname, firstname) if p))
```

Comment:
I know that they can be overriden, but we are talking about the default behaviour, that all agree that now is not the best choice and even annoying. A change in these methods with a migration script will handle it.